src/Shared/HttpSys/RequestProcessing/HeaderEncoding.cs

```diff
@@ -19,12 +16,7 @@ internal static unsafe string GetString(byte* pBytes, int byteCount, bool useLat
         }
         else
         {
-            return new ReadOnlySpan<byte>(pBytes, byteCount).GetAsciiOrUTF8StringNonNullCharacters(Encoding);
+            return new ReadOnlySpan<byte>(pBytes, byteCount).GetAsciiOrUTF8StringNonNullCharacters(Encoding.UTF8);
```

Comment:
Are these the same settings as Encoding.UTF8? I imagine there's a reason we had a custom instantiation.

Reply:
The original `Encoding` was also `UTF8`, but explicitly set `encoderShouldEmitUTF8Identifier: false` which is the only semantic difference from `Encoding.UTF8`. I was deferring to @jkotas's suggestion, see https://github.com/dotnet/aspnetcore/pull/44561#discussion_r996475101. Please let me know if you'd prefer to change it back.